op-batcher/batcher/service.go

```diff
@@ -94,6 +94,32 @@ func BatcherServiceFromCLIConfig(ctx context.Context, version string, cfg *CLICo
 	return &bs, nil
 }
 
+// union combines two slices of strings and returns a new slice with unique elements.
+func union(a, b []string) []string {
+	// Create a map to store unique elements
+	unique := make(map[string]struct{})
+
+	// Add elements from the first slice
+	for _, item := range a {
+		unique[item] = struct{}{}
+	}
+
+	// Add elements from the second slice
+	for _, item := range b {
+		unique[item] = struct{}{}
+	}
+
+	// Create a result slice to hold the unique elements
+	result := make([]string, 0, len(unique))
+
+	// Append the unique elements to the result slice
+	for item := range unique {
+		result = append(result, item)
+	}
+
+	return result
+}
```

Comment:
we could also put a more generic version into a new package `op-service/slices` for possible reuse...
```go
func Union[T comparable](a, b []T) []T {
    seen := make(map[T]struct{})
    var result []T

    for _, v := range a {
        if _, ok := seen[v]; !ok {
            seen[v] = struct{}{}
            result = append(result, v)
        }
    }

    for _, v := range b {
        if _, ok := seen[v]; !ok {
            seen[v] = struct{}{}
            result = append(result, v)
        }
    }

    return result
}
```
up to you.

Reply:
Nice idea, I did that here [bf58a0e](https://github.com/ethereum-optimism/optimism/pull/15603/commits/bf58a0e79a8fd81fa12637dfb82275283f27f346)